src/app/infra/security/jwt-service.ts

```diff
@@ -1,27 +1,55 @@
 import { env } from "@/app/env";
 import jwt from "jsonwebtoken";
 import { z } from "zod";
-import { OutputValidationError } from "../http/errors/output-validation/output-validation.error";
 
-const payloadOutputSchema = z.object({
+type PayloadOutput = {
+  sub: string;
+  username: string;
+  email: string;
+  exp: number;
+};
+
+type SanitizedPayload = jwt.JwtPayload & PayloadOutput;
+
+const payloadInputSchema = z.object({
   id: z.string(),
+  fullName: z.string(),
   username: z.string(),
   email: z.string(),
+});
+
+const payloadOutputSchema = z.object({
+  sub: z.string(),
+  username: z.string(),
+  email: z.string(),
+  iat: z.number(),
   exp: z.number(),
 });
 
-type Output = z.infer<typeof payloadOutputSchema>;
+export class JwtService {
+  static sign(payload: unknown): string {
+    const result = payloadInputSchema.safeParse(payload);
+    if (!result.success) {
+      throw new Error(
+        "Malformed payload: Payload has different data than expected."
+      );
+    }
 
-type PayloadOutput = jwt.JwtPayload & Output;
+    const sanitizedPayload = {
+      sub: result.data.id,
```

Comment:
`sub` is the convention name for `id` of the user in a JWT token.

You can check it here: https://jwt.io/ by pasting this token: `eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIwNWJkZTZlMS1iOWU0LTQ0M2QtODc4Zi0zZjkxM2Q0ZjQwNTciLCJmdWxsTmFtZSI6IkpvaG4gRG9lIiwidXNlcm5hbWUiOiJqZG9lIiwiZW1haWwiOiJqZG9lQGdtYWlsLmNvbSIsImlhdCI6MTc0NDkwMzI1NSwiZXhwIjoxNzQ0OTA2ODU1fQ.MckiT1UDsnLIt6rPcKK0QlXZf9m2Xo7JK7jCR9E04M8`

Then hover over the `sub` property (it means subject). A tooltip will be shown explaining.